pkg/controllers/dynakube/logmodule/daemonset/reconciler.go

```diff
@@ -0,0 +1,119 @@
+	serviceAccountName = "dynatrace-logmodule"
+)
+
+type Reconciler struct {
+	client    client.Client
+	apiReader client.Reader
+	dk        *dynakube.DynaKube
+}
+
+func NewReconciler(clt client.Client,
+	apiReader client.Reader,
+	dk *dynakube.DynaKube) *Reconciler {
+	return &Reconciler{
+		client:    clt,
+		apiReader: apiReader,
+		dk:        dk,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if !r.dk.NeedsLogModule() {
+		if meta.FindStatusCondition(*r.dk.Conditions(), conditionType) == nil {
+			return nil // no condition == nothing is there to clean up
+		}
+
+		query := daemonset.Query(r.client, r.apiReader, log)
+		err := query.Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: GetName(r.dk.Name), Namespace: r.dk.Namespace}})
+
+		if err != nil {
+			log.Error(err, "failed to clean-up LogModule config-secret")
+		}
+
+		meta.RemoveStatusCondition(r.dk.Conditions(), conditionType)
+
+		return nil // clean-up shouldn't cause a failure
+	}
+
+	ds, err := r.generateDaemonSet()
+	if err != nil {
+		return err
+	}
+
+	updated, err := daemonset.Query(r.client, r.apiReader, log).WithOwner(r.dk).CreateOrUpdate(ctx, ds)
+	if err != nil {
+		conditions.SetKubeApiError(r.dk.Conditions(), conditionType, err)
+
+		return err
+	}
+
+	if updated {
+		conditions.SetDaemonSetOutdated(r.dk.Conditions(), conditionType, GetName(r.dk.Name)) // needed to reset the timestamp
+		conditions.SetDaemonSetCreated(r.dk.Conditions(), conditionType, GetName(r.dk.Name))
+	}
+
+	return nil
+}
+
+func (r *Reconciler) generateDaemonSet() (*appsv1.DaemonSet, error) {
+	tenantUUID, err := r.dk.TenantUUIDFromConnectionInfoStatus()
+	if err != nil {
+		return nil, err
+	}
+
+	labels := k8slabels.NewCoreLabels(r.dk.Name, k8slabels.LogModuleComponentLabel)
+
+	maxUnavailable := intstr.FromInt(r.dk.FeatureOneAgentMaxUnavailable())
+
+	ds, err := daemonset.Build(r.dk, GetName(r.dk.Name), getContainer(*r.dk),
+		daemonset.SetInitContainer(getInitContainer(*r.dk)),
+		daemonset.SetAllLabels(labels.BuildLabels(), labels.BuildMatchLabels(), labels.BuildLabels(), r.dk.LogModuleTemplates().Labels),
+		daemonset.SetAllAnnotations(nil, r.dk.LogModuleTemplates().Annotations),
+		daemonset.SetServiceAccount(serviceAccountName),
+		daemonset.SetDNSPolicy(r.dk.LogModuleTemplates().DNSPolicy),
+		daemonset.SetAffinity(node.Affinity()),
+		daemonset.SetPriorityClass(r.dk.LogModuleTemplates().PriorityClassName),
+		daemonset.SetTolerations(r.dk.LogModuleTemplates().Tolerations),
+		daemonset.SetPullSecret(r.dk.ImagePullSecretReferences()...),
+		daemonset.SetUpdateStrategy(appsv1.DaemonSetUpdateStrategy{
+			RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+				MaxUnavailable: &maxUnavailable,
+			},
+		}),
+		daemonset.SetVolumes(getVolumes(r.dk.Name, tenantUUID)),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	err = hasher.AddAnnotation(ds)
+	if err != nil {
+		return nil, err
+	}
+
+	return ds, nil
+}
+
+func GetName(dkName string) string {
```

Comment:
we are in the `daemonset` package, that would be imported (probably with an alias) like `lmdaemonset`, then it would read `lmdaemonset.GetName`. So we avoid stuttering.

Reply:
the current usage _in_ the log module package confused me during review

we already have the name for other components in properties: https://github.com/Dynatrace/dynatrace-operator/blob/main/pkg/api/v1beta2/dynakube/properties.go#L100

Reply:
the fact that we did it a certain way doesn't make it correct 😅

the problem with putting everything in the properties that it just becomes a bloated mess (it already is)
- pretty sure the properties you brought up as example have been there since the oneagent-operator days
- there are several examples for not doing it like this:
   - https://github.com/Dynatrace/dynatrace-operator/blob/266389ab053979d6935dfd140e360b6b72ac8b3f/pkg/controllers/dynakube/deploymentmetadata/reconciler.go#L86-L89
   - https://github.com/Dynatrace/dynatrace-operator/blob/266389ab053979d6935dfd140e360b6b72ac8b3f/pkg/controllers/dynakube/processmoduleconfigsecret/reconciler.go#L213
   - https://github.com/Dynatrace/dynatrace-operator/blob/266389ab053979d6935dfd140e360b6b72ac8b3f/pkg/controllers/dynakube/proxy/reconciler.go#L128-L131
the problem with putting such things closest to where it's used could be that it's harder to discover
- you would have to look at the different packages to see, instead of the single big package
   - I personally find it more readable, that there is a `logmodule` package that has a subpackage `daemonset`, so if I want to get something `logmodule-daemonset` specific, I will just look at that package, and it will have no noise from the ActiveGate stuff, for example.

also, if we want to have each bigger section of the spec in their own package, then such properties like `OneAgentDaemonSetName` should no longer be in the properties:
- because, you would like a syntax like `dk.OneAgent().DaemonSetName()`, but this is not possible, as it would depend on the name of the `dk`, but the `DaemonSetName` method can't access that
  - you can do some tricks like, that each section has a hidden `ParentName` field that is not part of the yaml and is set in the `dk.OneAgent()` func for example


TLDR: The way we go about this is a bit inconsistent, but having everything in the properties IMO is not how we should do it, as it "doesn't scale", properties would be an ever growing package.

Reply:
done, moved it to the props for now